source/userguide/text_preparation_pipeline.rst

```diff
@@ -0,0 +1,661 @@
+
+.. code:: python
+
+    from mindmeld.text_preparation.tokenizers import LetterTokenizer
+
+    sentence_ja = "1年は365日"
+    letter_tokenizer = LetterTokenizer()
+    tokens = letter_tokenizer.tokenize(sentence_ja)
+    print([t['text'] for t in tokens])
+
+We see that the original text is split at the character level for non-latin characters but the number "365" remains as an unsegmented token.
+
+.. code:: python
+
+    ['1', '年', 'は', '365', '日']
+
+
+Spacy Tokenizer
+^^^^^^^^^^^^^^^
+The :attr:`SpacyTokenizer` splits up a sentence using `Spacy's language models <https://spacy.io/models>`_.
+Supported languages include English (en), Spanish (es), French (fr), German (de), Danish (da), Greek (el), Portuguese (pt), Lithuanian (lt), Norwegian Bokmal (nb), Romanian (ro), Polish (pl), Italian (it), Japanese (ja), Chinese (zh), Dutch (nl).
+If the required Spacy model is not already present it will automatically downloaded during runtime. 
+Let's use the :attr:`SpacyTokenizer` to tokenize the Japanese translation of "The gentleman is gone, no one knows why it happened!": 
+
+.. code:: python
+
+    from mindmeld.text_preparation.tokenizers import SpacyTokenizer
+    
+    sentence_ja = "紳士が過ぎ去った、 なぜそれが起こったのか誰にも分かりません！"
+    spacy_tokenizer_ja = SpacyTokenizer(language="ja", spacy_model_size="lg")
+    tokens = spacy_tokenizer_ja.tokenize(sentence_ja)
+
+We see that the original text is split semantically and not simply by whitespace.
+
+.. code:: python
+
+    ['紳士', 'が', '過ぎ', '去っ', 'た', '、', 'なぜ', 'それ', 'が', '起こっ', 'た', 'の', 'か', '誰', 'に', 'も', '分かり', 'ませ', 'ん', '！']
+
+
+Creating a Custom Tokenizer
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This section includes boilerplate code to build a custom tokenizer class. Let's rebuild a :attr:`CharacterTokenizer` class that creates a token for each character in a string as long as the
+character is not a space. A custom tokenizer must extend from MindMeld's abstract :attr:`Tokenizer` class:
+
+
+.. code:: python
+
+    from abc import ABC, abstractmethod
+
+    class Tokenizer(ABC):
+        """Abstract Tokenizer Base Class."""
+
+        @abstractmethod
+        def tokenize(self, text):
+            """
+            Args:
+                text (str): The text to tokenize.
+            Returns:
+                tokens (List[Dict]): List of tokenized tokens which a represented as dictionaries.
+                    Keys include "start" (token starting index), and "text" (token text).
+                    For example: [{"start": 0, "text":"hello"}]
+            """
+            raise NotImplementedError("Subclasses must implement this method")
+
+
+Note that any MindMeld tokenizer must return the final tokens as a list of dictionaries. Where each dictionary represents a single token and contains the "start" index of the token and the "text" of the token.
+Here is an example of the expected output for the tokens generated when tokenizing the phrase "Hi Andy": [{"start": 0, "text":"Hi"}, {"start": 3, "text":"Andy"}].
```

Comment:
a point of clarity: if the preprocessor changes the string, are these indices referencing the raw text or the processed text?

Reply:
Good point, these indices reference the processed text. I'll add this information here